chains/avalanche/src/lib.rs

```diff
@@ -0,0 +1,286 @@
+//! - `rosetta_config_ethereum`: Configuration for Ethereum Rosetta server.
+//! - `rosetta_server_avalanche`: Custom client implementation for interacting with Avalanche.
+//! - `sha3`: SHA-3 (Keccak) implementation for hashing.
+//! - `tokio`: Asynchronous runtime for running async functions.
+//!
+//! ## Usage
+//!
+//! To run the tests, execute the following command:
+//!
+//! ```sh
+//! cargo test --package rosetta-testing-avalanche --lib -- tests --nocapture
+//! ```
+//!
+//! Note: The code assumes a local Avalanche Nitro Testnet node running on `ws://127.0.0.1:8548` and
+//! a local Ethereum node on `http://localhost:8545`. Ensure that these endpoints are configured correctly.
+
+#[allow(clippy::ignored_unit_patterns, clippy::pub_underscore_fields)]
+#[cfg(test)]
+mod tests {
+    use alloy_sol_types::{sol, SolCall};
+    use anyhow::Result;
+    use ethers::types::H256;
+    use ethers_solc::{artifacts::Source, CompilerInput, EvmVersion, Solc};
+    use hex_literal::hex;
+    use rosetta_client::Wallet;
+    use rosetta_config_ethereum::{AtBlock, CallResult};
+    use rosetta_core::BlockchainClient;
+    use rosetta_server_ethereum::MaybeWsEthereumClient;
+    use serial_test::serial;
+    use sha3::Digest;
+    use std::{collections::BTreeMap, future::Future, path::Path};
+
+    /// Account used to fund other testing accounts.
+    const FUNDING_ACCOUNT_PRIVATE_KEY: [u8; 32] =
+        hex!("d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d");
+
+    /// Avalanche rpc url
+    const AVALANCHE_RPC_WS_URL: &str = "ws://127.0.0.1:9650/ext/bc/localnew/ws";
+
+    sol! {
+        interface TestContract {
+            event AnEvent();
+            function emitEvent() external;
+
+            function identity(bool a) external view returns (bool);
+        }
+    }
+
+    /// Run the test in another thread while sending txs to force binance to mine new blocks
+    /// # Panic
+    /// Panics if the future panics
+    async fn run_test<Fut: Future<Output = ()> + Send + 'static>(future: Fut) {
+        // Guarantee that only one test is incrementing blocks at a time
+        static LOCK: tokio::sync::Mutex<()> = tokio::sync::Mutex::const_new(());
+
+        // Run the test in another thread
+        let test_handler = tokio::spawn(future);
+
+        // Acquire Lock
+        let guard = LOCK.lock().await;
+
+        // Check if the test is finished after acquiring the lock
+        if test_handler.is_finished() {
+            // Release lock
+            drop(guard);
+
+            // Now is safe to panic
+            if let Err(err) = test_handler.await {
+                std::panic::resume_unwind(err.into_panic());
+            }
+            return;
+        }
+
+        // Now is safe to panic
+        if let Err(err) = test_handler.await {
+            // Resume the panic on the main task
+            std::panic::resume_unwind(err.into_panic());
+        }
+    }
```

Comment:
I see we have this exact same code copied between 3 different crates now, can you create an crate to put out this shared testing logic? otherwise if we change it, we will need to change it in 3 different places now.

Reply:
Can I do it in different PR

Reply:
I will fix it in https://github.com/Analog-Labs/chain-connectors/issues/257